src/main/java/io/cloudsoft/terraform/TerraformSshDriver.java

```diff
@@ -151,20 +156,38 @@ public void customize() {
                                 "if grep -q \"No errors detected\" <<< $(unzip -t "+ cfgPath +" ); then "
                                         + "mv " + cfgPath + " " + cfgPath + ".zip && cd " + getRunDir() + " &&"
                                          + "unzip " + cfgPath + ".zip ; fi")
+                        .requiringExitCodeZero()
                         .environmentVariables(getShellEnvironment())
                         .summary("Preparing configuration (unzip of necessary)...")
-                        .requiringExitCodeZero().newTask()
+                        .newTask()
                         .asTask())
                 .add(SshTasks.newSshExecTaskFactory(getMachine(), initCommand())
+                        .requiringZeroAndReturningStdout()
                         .environmentVariables(getShellEnvironment())
                         .summary("Initializing terraform infrastructure")
-                        .requiringZeroAndReturningStdout().newTask()
+                        .newTask()
                         .asTask())
                 .build());
         DynamicTasks.waitForLast();
 
-        if (initTask.asTask().isError()) {
```

Comment:
Don't we want to check this as well?

Reply:
@algairim I realized that code is never reached. When a task fails and exit code zero is requested, a runtime exception is thrown and propagated.